pkg/discovery/network/generic.go

```diff
@@ -62,13 +58,112 @@ func discoverGenericNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, er
 }
 
 func findClusterIPRange(clientSet kubernetes.Interface) (string, error) {
+	clusterIPRange, err := findClusterIPRangeApiserver(clientSet)
+	if err != nil || clusterIPRange != "" {
+		return clusterIPRange, err
+	}
+
+	clusterIPRange, err = findClusterIPRangeSvcCreation(clientSet)
+	if err != nil || clusterIPRange != "" {
+		return clusterIPRange, err
+	}
+
+	return "", nil
+}
+
+func findClusterIPRangeApiserver(clientSet kubernetes.Interface) (string, error) {
 	return findPodCommandParameter(clientSet, "component=kube-apiserver", "--service-cluster-ip-range")
 }
 
+func findClusterIPRangeSvcCreation(clientSet kubernetes.Interface) (string, error) {
+	// find service cidr based on https://stackoverflow.com/questions/44190607/how-do-you-find-the-cluster-service-cidr-of-a-kubernetes-cluster
+	invalidSvcSpec := &v1.Service{
+		ObjectMeta: v1meta.ObjectMeta{
+			Name: "invalid-svc",
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "1.1.1.1",
+			Ports: []v1.ServicePort{
+				{
+					Port: 443,
+					TargetPort: intstr.IntOrString{
+						IntVal: 443,
+					},
+				},
+			},
+		},
+	}
+	_, err := clientSet.CoreV1().Services("submariner-operator").Create(invalidSvcSpec)
+
+	// creating invalid svc didn't fail as expected
+	if err == nil {
+		return "", nil
+	}
+
+	return parseToServiceCidr(err.Error())
+}
+
+func parseToServiceCidr(msg string) (string, error) {
+	// expected msg is below:
+	//   "The Service \"invalid-svc\" is invalid: spec.clusterIPs: Invalid value: []string{\"1.1.1.1\"}:
+	//   failed to allocated ip:1.1.1.1 with error:provided IP is not in the valid range.
+	//   The range of valid IPs is 10.45.0.0/16"
+	// expected matched string is below:
+	//   10.45.0.0/16
+	re := regexp.MustCompile(".*valid IPs is (.*)$")
+
+	match := re.FindStringSubmatch(msg)
+	if match == nil {
+		return "", fmt.Errorf("parsing (%s) failed. it doesn't match with %v", msg, re)
```

Comment:
We don't return an error above if no error was returned from service creation so we should be consistent here. I think that's the correct behavior as it means we didn't find the desired data. As above I think we should log a message:

`status.QueueWarningMessage(fmt.Sprintf"Could not determine the service IP range via service creation - the expected error was not returned. The actual error was %q", msg))`